src/twisted/internet/task.py

```diff
@@ -817,8 +817,14 @@ def advance(self, amount):
         @param amount: The number of seconds which to advance this clock's
         time.
         """
-        self.rightNow += amount
         self._sortCalls()
+        if amount is not None:
+            self.rightNow += amount
+        else:
+            if self.calls:
+                self.rightNow = max(self.calls[0].getTime(), self.rightNow)
+            else:
+                return
```

Comment:
This should be in a separate PR, and probably exist as a separate function, rather than using a default argument.